helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java

```diff
@@ -188,11 +187,7 @@ public static boolean hasErrorPartitions(HelixDataAccessor dataAccessor, String
     if (liveInstance != null) {
       String sessionId = liveInstance.getEphemeralOwner();
 
-      PropertyKey currentStatesKey = propertyKeyBuilder.currentStates(instanceName, sessionId);
-      List<String> resourceNames = dataAccessor.getChildNames(currentStatesKey);
-      for (String resourceName : resourceNames) {
-        PropertyKey key = propertyKeyBuilder.currentState(instanceName, sessionId, resourceName);
-
+      for (PropertyKey key : getAllCurrentStateKeys(dataAccessor, instanceName, sessionId)) {
```

Comment:
This is not necessary to instance validation util. Because this management API is used for check real data resource. Our split could even better to filter out the TASK current state.

Reply:
@dasahcc removed